internal/datastore/memdb/revisions.go

```diff
@@ -70,6 +70,15 @@ func (mdb *memdbDatastore) checkRevisionLocal(revisionRaw revision.Decimal) erro
 	now := revisionFromTimestamp(time.Now().UTC())
 
 	if revisionRaw.GreaterThan(now) {
+		headRevision, err := mdb.headRevision()
```

Comment:
Hmm, I don't really like this approach. Wouldn't a better approach be to check that it's "< 1 big part more" than what was already passed back? This seems like it will generate a +2 snapshot when we don't need it...

Reply:
Not sure I follow... its not generating a new revision, simply check that it is less than or equal to the current head revision

Reply:
Why not just always use the `mdb.HeadRevision` here?

Reply:
It was returning a slightly different data type, if I recall correctly

Reply:
Or do you mean why not just use `headRevision` always?

Reply:
Yeah I mean in this code block, if you're calling out to headRevision if it's > now, why not just always use the headrevision path.

Reply:
I was trying to avoid hitting the lock condition unless absolutely necessary; this way, we only hit the lock on the check revision when necessary